services/supervisor/tests/supervisor/config.rs

```diff
@@ -0,0 +1,267 @@
+                );
+            }
+        }
+    }
+}
+
+fn assert_config_change_entry_is_empty(testkit: &TestKit) {
+    let snapshot = testkit.snapshot();
+    assert!(!Schema::new("supervisor", &snapshot).config_propose_entry().exists());
+}
+
+
+fn prepare_config_proposal(new_consensus_config: &ConsensusConfig, change_height: Height) -> (ConfigPropose, Hash) {
+    let config_proposal = ConfigPropose{
+        actual_from: change_height,
+        changes: vec!(ConfigChange::Consensus(new_consensus_config.clone())),
+    };
+    let proposal_hash = config_proposal.object_hash();
+
+    (config_proposal, proposal_hash)
+}
+
+#[test]
+fn test_add_nodes_to_validators() {
+    let mut testkit = TestKitBuilder::auditor()
+        .with_validators(1)
+        .with_service(Supervisor)
+        .create();
+
+    let cfg_change_height = Height(5);
+    let new_consensus_config = {
+        let mut cfg = testkit.consensus_config();
+        // Add us node.
+        cfg.validator_keys.push(testkit.us().public_keys());
+        // Add new node.
+        cfg.validator_keys
+            .push(testkit.network_mut().add_node().public_keys());
+        cfg
+    };
+
+    let (new_config_propose, _) = prepare_config_proposal(&new_consensus_config, cfg_change_height);
+
+    testkit.create_block_with_transaction(
+        testkit.sign_config_propose(new_config_propose)
+    );
+
+    testkit.create_blocks_until(cfg_change_height);
+
+    assert_config_change_entry_is_empty(&testkit);
+    assert_eq!(testkit.network().us().validator_id(), Some(ValidatorId(1)));
+    assert_eq!(&testkit.network().validators()[1], testkit.network().us());
+    assert_eq!(testkit.consensus_config(), new_consensus_config);
+}
+
+#[test]
+fn exclude_us_from_validators() {
+    let mut testkit = TestKitBuilder::validator()
+        .with_validators(4)
+        .with_service(Supervisor)
+        .create();
+
+    let cfg_change_height = Height(5);
+    let new_consensus_config = {
+        let mut cfg = testkit.consensus_config();
+        // Remove us from validators
+        cfg.validator_keys.remove(0);
+        cfg
+    };
+
+    let old_validators = testkit.network().validators();
+
+    let (config_proposal, proposal_hash) = prepare_config_proposal(&new_consensus_config, cfg_change_height);
+
+    testkit.create_block_with_transaction(
+        testkit.sign_config_propose(config_proposal)
+    );
+
+    testkit.nodes_vote_for_proposed_config(proposal_hash, ValidatorId(0));
+
+    testkit.create_blocks_until(cfg_change_height.next());
+
+    let new_validators = testkit.network().validators();
+
+    assert_config_change_entry_is_empty(&testkit);
+    assert_eq!(testkit.consensus_config(), new_consensus_config);
+    assert_eq!(testkit.network().us().validator_id(), None);
+    assert_ne!(old_validators, new_validators);
+}
+
+#[test]
+fn exclude_other_from_validators() {
+    let mut testkit = TestKitBuilder::validator()
+        .with_validators(2)
```

Comment:
Testing BFT-inspired algorithm with 2 validators (and even 1) seems strage to me.